server/src/index.ts

```diff
@@ -94,6 +95,13 @@ const createTransport = async (req: express.Request): Promise<Transport> => {
 
     console.log("Connected to SSE transport");
     return transport;
+  } else if (transportType === "streamable-http") {
+    const transport = new StreamableHTTPClientTransport(
```

Comment:
To Auth feature works you should send headers to the server

```suggestion
    const STREAMABLE_HTTP_HEADERS_PASSTHROUGH = ["authorization"];

    const headers: HeadersInit = {};

    for (const key of STREAMABLE_HTTP_HEADERS_PASSTHROUGH) {
      if (req.headers[key] === undefined) {
        continue;
      }

      const value = req.headers[key];
      headers[key] = Array.isArray(value) ? value[value.length - 1] : value;
    }
  
    const transport = new StreamableHTTPClientTransport(
```

Reply:
Thanks. I have added the recommended changes and tested them. They work well.

**MCP Inspect**
![Screenshot 2025-04-17 at 9 28 21 PM](https://github.com/user-attachments/assets/c0c5e8cc-af30-4930-b3f9-84ca10e0a21a)

**Req/Resp Headers**
![Screenshot 2025-04-17 at 9 27 49 PM](https://github.com/user-attachments/assets/e37cbdbc-f2b4-48c5-ab01-5f09f8de99f5)
![Screenshot 2025-04-17 at 9 28 02 PM](https://github.com/user-attachments/assets/627fab23-5c09-4b8e-ae91-07df3f9209a2)